From: Jeff Layton <jlayton@redhat.com>
Date: Thu, 26 Jan 2017 16:14:18 -0500
Subject: ceph: don't update_dentry_lease unless we actually got one
Git-commit: 80d025ffede88969f6adf7266fbdedfd5641148a
Patch-mainline: v4.11-rc1
References: FATE#322288

This if block updates the dentry lease even in the case where
the MDS didn't grant one.

Signed-off-by: Jeff Layton <jlayton@redhat.com>
Reviewed-by: Yan, Zheng <zyan@redhat.com>
Signed-off-by: Ilya Dryomov <idryomov@gmail.com>
Acked-by: Luis Henriques <lhenriques@suse.com>
---
 fs/ceph/inode.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/fs/ceph/inode.c b/fs/ceph/inode.c
index 35a8c453bea6..542e3c6b0f1b 100644
--- a/fs/ceph/inode.c
+++ b/fs/ceph/inode.c
@@ -1286,8 +1286,8 @@ int ceph_fill_trace(struct super_block *sb, struct ceph_mds_request *req)
 				ceph_dir_clear_ordered(dir);
 				dout("d_delete %p\n", dn);
 				d_delete(dn);
-			} else {
-				if (have_lease && d_unhashed(dn))
+			} else if (have_lease) {
+				if (d_unhashed(dn))
 					d_add(dn, NULL);
 				update_dentry_lease(dn, rinfo->dlease,
 						    session,

